src/app/api/user/page_views/tool.ts

```diff
@@ -0,0 +1,83 @@
+import prisma from "@/lib/prisma";
+
+export async function getMarkedStories(
+  userId: string,
+  storyIds: string[],
+  type: "BOOK_MARK" | "BRAIN",
+) {
+  const result: { [key: string]: boolean } = {};
+  let storiesFound: { storyId: string }[] = [];
+  if (type === "BOOK_MARK") {
+    storiesFound = await prisma.bookmark.findMany({
+      where: {
+        userId: userId,
+        storyId: {
+          in: storyIds,
+        },
+      },
+      select: {
+        storyId: true,
+      },
+    });
+  } else if (type === "BRAIN") {
+    storiesFound = await prisma.brain.findMany({
+      where: {
+        userId: userId,
+        storyId: {
+          in: storyIds,
+        },
+      },
+      select: {
+        storyId: true,
+      },
+    });
+  }
+  storiesFound.forEach((story) => {
+    result[story.storyId] = true;
+  });
+  return result;
+}
+
+export async function checkDifferentDateRead(
+  storyId: string,
+  userId: string,
+  timeZone: number,
+) {
+  const UTCTime = new Date();
+  let isNewDay = false;
+  const lastPost = await prisma.pageView.findFirst({
+    where: {
+      userId,
+      storyId,
+    },
+    orderBy: {
+      createdAt: "desc",
+    },
+    select: {
+      createdAt: true,
+      id: true,
+    },
+  });
+  if (!lastPost) {
+    throw new Error("lastPost not found in isDifferentDateRead function");
+  }
+  const lastUTCTime = lastPost.createdAt;
+  const DAY_IN_MILISECOND = 86400000;
+
+  if (UTCTime.getTime() - lastUTCTime.getTime() >= DAY_IN_MILISECOND) {
+    isNewDay = true;
+  } else {
```

Comment:
For some reason, in my testing, this sometimes returns the wrong value.  Would it be possible to discuss the way this works, and maybe figure out why my tests are not working?